pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -1280,6 +1280,9 @@ class Timestamp(_Timestamp):
         elif not is_offset_object(freq):
             freq = to_offset(freq)
 
+        naive_ts = convert_to_tsobject(ts_input, None, unit, 0, 0, nanosecond or 0)
+        tz_localize_to_utc_single(naive_ts.value, ts.tzinfo, False, "raise")
```

Comment:
this is a super hotspot, if we *really* need to do this then need to confirm performance.